third_party/2and3/werkzeug/test.pyi

```diff
@@ -67,28 +70,63 @@ class EnvironBuilder:
 
 class ClientRedirectError(Exception): ...
 
-class Client:
+_ResponseWrapper = TypeVar('_ResponseWrapper', bound=Response)
+
+class Client(Generic[_ResponseWrapper]):
     application: Any
-    response_wrapper: Any
+    response_wrapper: Type[_ResponseWrapper]
     cookie_jar: Any
     allow_subdomain_redirects: Any
-    def __init__(self, application, response_wrapper: Optional[Any] = ..., use_cookies: bool = ...,
+    def __init__(self, application, response_wrapper: Optional[Type[_ResponseWrapper]] = ..., use_cookies: bool = ...,
                  allow_subdomain_redirects: bool = ...): ...
     def set_cookie(self, server_name, key, value: str = ..., max_age: Optional[Any] = ..., expires: Optional[Any] = ...,
                    path: str = ..., domain: Optional[Any] = ..., secure: Optional[Any] = ..., httponly: bool = ...,
                    charset: Text = ...): ...
     def delete_cookie(self, server_name, key, path: str = ..., domain: Optional[Any] = ...): ...
     def run_wsgi_app(self, environ, buffered: bool = ...): ...
     def resolve_redirect(self, response, new_location, environ, buffered: bool = ...): ...
-    def open(self, *args, **kwargs): ...
-    def get(self, *args, **kw): ...
-    def patch(self, *args, **kw): ...
-    def post(self, *args, **kw): ...
-    def head(self, *args, **kw): ...
-    def put(self, *args, **kw): ...
-    def delete(self, *args, **kw): ...
-    def options(self, *args, **kw): ...
-    def trace(self, *args, **kw): ...
+
+    @overload
+    def open(self, *args, as_tuple: Literal[True], **kwargs) -> Tuple[Mapping[str, Any], _ResponseWrapper]: ...
```

Comment:
Unfortunately, this functions and the ones below only return a `_ResponseWrapper` instance, if that is set. In case it is not set (and therefore `None`), it will return a triple instead of a `_ResponseWrapper` instance. I think the best we can do is to return `Any` instead of `_ResponseWrapper`.

`wsgiref.types` has an alias `WSGIEnvironment` for `Dict[str, Any]`. Generally, WSGI environment are mutable (per [PEP 3333](https://www.python.org/dev/peps/pep-3333/)).

(Same for functions below.)

Reply:
Ouch you're right. I hadn't cheked what was the type in case no wrapper was set. But actually would it be OK to consider the default `_ResponseWrapper` to be `Tuple[...]` and thus keeping the strong typing here? I've removed the bound for the typevar.

Done for WSGIEnv.